cli/command/defaultcontextstore.go

```diff
@@ -0,0 +1,182 @@
+	store.Store
+	Resolver DefaultContextResolver
+}
+
+// resolveDefaultContext creates a ContextMetadata for the current CLI invocation parameters
+func resolveDefaultContext(opts *cliflags.CommonOptions, config *configfile.ConfigFile, stderr io.Writer) (*DefaultContext, error) {
+	stackOrchestrator, err := GetStackOrchestrator("", "", config.StackOrchestrator, stderr)
+	if err != nil {
+		return nil, err
+	}
+	contextTLSData := store.ContextTLSData{
+		Endpoints: make(map[string]store.EndpointTLSData),
+	}
+	contextMetadata := store.ContextMetadata{
+		Endpoints: make(map[string]interface{}),
+		Metadata: DockerContext{
+			Description:       "",
+			StackOrchestrator: stackOrchestrator,
+		},
+		Name: DefaultContextName,
+	}
+
+	dockerEP, err := resolveDefaultDockerEndpoint(opts)
+	if err != nil {
+		return nil, err
+	}
+	contextMetadata.Endpoints[docker.DockerEndpoint] = dockerEP.EndpointMeta
+	if dockerEP.TLSData != nil {
+		contextTLSData.Endpoints[docker.DockerEndpoint] = *dockerEP.TLSData.ToStoreTLSData()
+	}
+
+	// Default context uses env-based kubeconfig for Kubernetes endpoint configuration
+	kubeconfig := os.Getenv("KUBECONFIG")
+	if kubeconfig == "" {
+		kubeconfig = filepath.Join(homedir.Get(), ".kube/config")
+	}
+	kubeEP, err := kubernetes.FromKubeConfig(kubeconfig, "", "")
+	if stackOrchestrator == OrchestratorKubernetes && err != nil {
+		return nil, errors.Wrap(err, "default orchestrator is kubernetes but kubernetes endpoint could not be found")
+	}
+	if err == nil {
+		contextMetadata.Endpoints[kubernetes.KubernetesEndpoint] = kubeEP.EndpointMeta
+		if kubeEP.TLSData != nil {
+			contextTLSData.Endpoints[kubernetes.KubernetesEndpoint] = *kubeEP.TLSData.ToStoreTLSData()
+		}
+	}
+
+	return &DefaultContext{Meta: contextMetadata, TLS: contextTLSData}, nil
+}
+
+// ListContexts implements store.Store's ListContexts
+func (s *ContextStoreWithDefault) ListContexts() ([]store.ContextMetadata, error) {
+	contextList, err := s.Store.ListContexts()
+	if err != nil {
+		return nil, err
+	}
+	defaultContext, err := s.Resolver()
+	if err != nil {
+		return nil, err
+	}
+	return append(contextList, defaultContext.Meta), nil
+}
+
+// CreateOrUpdateContext is not allowed for the default context and fails
+func (s *ContextStoreWithDefault) CreateOrUpdateContext(meta store.ContextMetadata) error {
+	if meta.Name == DefaultContextName {
+		return errors.New("default context and cannot be created nor updated")
+	}
+	return s.Store.CreateOrUpdateContext(meta)
+}
+
+// RemoveContext is not allowed for the default context and fails
+func (s *ContextStoreWithDefault) RemoveContext(name string) error {
+	if name == DefaultContextName {
+		return errors.New("default context and cannot be removed")
+	}
+	return s.Store.RemoveContext(name)
+}
+
+// GetContextMetadata implements store.Store's GetContextMetadata
+func (s *ContextStoreWithDefault) GetContextMetadata(name string) (store.ContextMetadata, error) {
+	if name == DefaultContextName {
```

Comment:
Not sure about this, but should we also accept an empty name here ?

Reply:
Then maybe add a function `isDefaultContext` which resolves on `default` and on empty string? And use it in the other methods `RemoveContext` and `CreateOrUpdateContext` ?

Reply:
Empty string is not a valid context name anymore. When not defined, the context name falls back to `default`